.github/workflows/prepare_release.yml

```diff
@@ -48,12 +48,26 @@ jobs:
           string: ${{ matrix.provider }}
       - name: Check out project sources
         uses: actions/checkout@v3
+      - name: Check if publication to Maven Central is necessary
+        id: check-for-release
+        shell: python
+        run: |
+          import os
+          import json
+          with open("src/main/resources/version-config.json") as version_config:
+            schemas = json.load(version_config)
+            version = [schema["kotlinVersion"] for schema in schemas if schema["providerName"] == "${{ matrix.provider }}"][0]
+            is_release = not version.endswith("-SNAPSHOT")
+            with open(os.environ['GITHUB_OUTPUT'], 'a') as fh:
+              fh.write('is_release=' + str(is_release).lower())        
```

Comment:
We won't have time for that now probably, but I'd prefer this to be implemented as a separate Gradle task in `buildSrc`. 

We would get:

- encapsulation — no tool operates on `version-config.json` directly besides Gradle scripts — easier to test and maintain stuff.
- consistency — one way of working with `version-config.json`.

I've created an issue to explore that option: https://github.com/VirtuslabRnD/pulumi-kotlin/issues/261

Reply:
Yeah, good point